keychain/src/extkey_bip32.rs

```diff
@@ -89,27 +89,34 @@ pub trait BIP32Hasher {
 }
 
 /// Implementation of the above that uses the standard BIP32 Hash algorithms
+#[derive(Clone, Debug)]
 pub struct BIP32GrinHasher {
+	is_floo: bool,
 	hmac_sha512: Hmac<Sha512>,
 }
 
 impl BIP32GrinHasher {
 	/// New empty hasher
-	pub fn new() -> BIP32GrinHasher {
+	pub fn new(is_floo: bool) -> BIP32GrinHasher {
 		BIP32GrinHasher {
+			is_floo: is_floo,
 			hmac_sha512: HmacSha512::new(GenericArray::from_slice(&[0u8; 128])),
 		}
 	}
 }
 
 impl BIP32Hasher for BIP32GrinHasher {
-	fn network_priv() -> [u8; 4] {
-		// gprv
-		[0x03, 0x3C, 0x04, 0xA4]
+	fn network_priv(&self) -> [u8; 4] {
+		match self.is_floo {
+			true => [0x03, 0x27, 0x3A, 0x10],  // fprv
+			false => [0x03, 0x3C, 0x04, 0xA4], // gprv
+		}
 	}
-	fn network_pub() -> [u8; 4] {
-		// gpub
-		[0x03, 0x3C, 0x08, 0xDF]
+	fn network_pub(&self) -> [u8; 4] {
+		match self.is_floo {
+			true => [0x03, 0x27, 0x3E, 0x4A],  // fpub
```

Comment:
This seems a bit off for very low values.
```
>>> base58.encode(b'\x03\x27\x3e\x4a' + b'\x00' * 78) 
'fpuazZrso2XbHqNSnWB8m29iu12gwj4N6c8yRN9YqCzaW49pp5qd3SZVRci7jptrCgxw7L3Gs3kA7KSiz8jainXM7ENu228yFQ4nKzU82GWkDjd'
>>> base58.encode(b'\x03\x27\x3e\x4a' + b'\xff' * 78)
'fpub8sZZXw2wbqVpTV8TYeHJwGk4xijorUc2dSpjKLdGMzREmmmF8J4q4K1W4goyipXrXo3QUXeGV4gyDTgSzS2eBcZbYyfyGZzxzrK7Lb4fGN2'
```
If you change it to 0x03273e4b seems to work
```
>>> base58.encode(b'\x03\x27\x3e\x4b' + b'\x00' * 78) 
'fpub8sZZXw2wbqVpTV8TYeHJwGk4xijorUc2dSpjKLdGMzREmmmF8J4q4K1W4goyipXrXo3QUXeGV4gyDTgSzS2eBcZbYyfyGZzxzrK7Lb4fGN3'
>>> base58.encode(b'\x03\x27\x3e\x4b' + b'\xff' * 78)
'fpubHBGFGqYHuqdC8U5nLGQtyYTSyiRFcM55qXVuoUFxDvgejTgsD9aAh1JtPYj7Ex6mxG3Y61YNrowDSndKG5XwFzkJ5wCyHjw9fiA6eucaJzS'
```